Add a separate admin dashboard

margaretoti opened the pull request:
Admins are now directed to a separate dashboard (in view `admin/books/index.html.erb`) where they can add, edit, and delete books. 

This removes the admin logic from the view `books_controller/index.html.erb` and also means that separate patron tests to check that patrons cannot add, edit, or delete books are no longer needed.

The `yaml` file has also been updated to indicate admin specific links and text.

margaretoti committed:
Add a separate admin dashboard

Admins are now directed to a separate dashboard where they can add,
edit, and delete books. This removes the admin logic from the
books_controller/index and also means that separate patron tests to
check that admins cannot add, edit, or delete books are no longer
needed.

paulcsmith commented on config/routes.rb:
```diff
@@ -15,6 +15,6 @@
 
   namespace :admin do
-    resources :books, only: [:new, :edit, :create, :update, :destroy]
+    resources :books
```
I think that a `show` route is not used so you may want to still use `only`

paulcsmith commented on spec/features/admin_deletes_book_spec.rb:
```diff
@@ -19,15 +19,3 @@ def delete_book
-    expect(page).to have_content(t('books.index.header'))
-    expect(page).not_to have_content(t('books.delete.link'))
-  end
-end
```
Wo hoo!

paulcsmith reviewed (commented):
Very nice PR/commit description!

There is a small problem in that the tests are still testing the *old* implementation and passing. They should be failing, but they're not because the old implementation is still there.

You'd also need to change the existing tests to point to the new admin books index. LMK when that's done and I'll re-review!

margaretoti committed:
Direct admins and non-admins to correct controller

margaretoti committed:
Restructure yaml file

Restructured yaml file to have admin specific links and text. All other
text lives under books as opposed to admin/books.

margaretoti committed:
Redirect admin actions to admin paths

margaretoti commented on app/controllers/admin/books_controller.rb:
```diff
@@ -17,7 +17,7 @@ def create
     if @book.save
       flash[:success] = t('admin.books.create.flash_success')
-      redirect_to books_path
+      redirect_to admin_books_path
```
@paulcsmith. My tests all pass now, but when I investigate behavior in the browser, the behavior is not what I expect. Any advice you have on how to solve this problem would be greatly appreciated. Thanks!

paulcsmith commented on config/locales/en.yml:
```diff
@@ -1,25 +1,29 @@
+        flash_success: Book edited
+        button: Update Book
+      validations:
+        required: "can\'t be blank"
```
I don't think you need the backslash here since it's already in `""`

paulcsmith reviewed (approved):
Looks great! Just one small comment
